integration-tests/tests/src/node/ssl.ts

```diff
@@ -0,0 +1,253 @@
+    const onError: ErrorCallback = () => (syncErrorCallbackInvoked = true);
+
+    const realm = await openRealm(this.app.currentUser, ssl, onError);
+
+    expect(realm.syncSession?.isConnected()).to.be.true;
+    expect(realm.syncSession?.config.ssl).to.deep.equal({ validate: false, certificatePath: undefined });
+    expect(validationCallbackInvoked).to.be.false;
+    expect(syncErrorCallbackInvoked).to.be.false;
+
+    closeRealm(realm);
+  });
+
+  it("does not connect when rejecting the server's SSL certificate", async function (this: RealmContext) {
+    let validationCallbackInvoked = false;
+    const ssl: SSLConfiguration = {
+      validate: true,
+      certificatePath: undefined,
+      validateCertificates: () => {
+        validationCallbackInvoked = true;
+        // Reject the certificate.
+        return false;
+      },
+    };
+
+    const onErrorHandle = createPromiseHandle();
+    const onError: ErrorCallback = (_, error) => {
+      const SSL_SERVER_CERT_REJECTED = 117;
+      if (error.code === SSL_SERVER_CERT_REJECTED) {
+        onErrorHandle.resolve();
+      } else {
+        onErrorHandle.reject(`Expected the error to be an SSL server certificate rejection, got: ${error.message}`);
+      }
+    };
+
+    await expect(openRealm(this.app.currentUser, ssl, onError)).to.be.rejectedWith("SSL server certificate rejected");
+    expect(validationCallbackInvoked).to.be.true;
+    await expect(onErrorHandle.promise).to.not.be.rejected;
+  });
+
+  it("verifies the server's SSL certificate", async function (this: RealmContext) {
+    let validationCallbackInvoked = false;
+    const ssl: SSLConfiguration = {
+      validate: true,
+      certificatePath: undefined,
+      validateCertificates: (verifyObject: SSLVerifyObject) => {
+        validationCallbackInvoked = true;
+
+        // The `depth` of the root certificate will be the length of the chain - 1. All
+        // certificates between the root and the actual server certificate have a depth > 0.
+        const isServerCertificate = verifyObject.depth === 0;
+        const x509 = new X509Certificate(verifyObject.pemCertificate);
+
+        if (verifyObject.acceptedByOpenSSL) {
+          // If it's the actual server certificate, Core recommends always checking the host;
+          // otherwise, if it's higher up in the chain and accepted by OpenSSL, they recommend
+          // returning `true`.
+          return isServerCertificate ? !!x509.checkHost(verifyObject.serverAddress) : true;
+        } else {
+          // If the certificate is not accepted by OpenSSL, Core recommends using an independent
+          // verification step. That step is represented here by checking the dates.
+          // (TODO: Add extra validation)
+          const now = new Date();
+          const isValid = now >= new Date(x509.validFrom) && now <= new Date(x509.validTo);
+          return isServerCertificate ? isValid && !!x509.checkHost(verifyObject.serverAddress) : isValid;
```

Comment:
As mentioned in the comment, I'm currently using the `validFrom/To` of the certificate as the extra validation step.

Not sure if the following should pass but it returns `false` for all certificates passed to this callback:
```ts
// Verify that the certificate was signed by the given public key.
const verified = x509.verify(x509.publicKey);
```

If comparing against e.g. `x509.fingerprint256` or another known certificate, we'd had to know where those are so that we can hardcode it here, but I haven't found any info on that.